### PR TITLE
PUD-1201: Include path and field name separately

### PR DESCRIPTION
**Pull request by philsegal**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/service/RecallAuditService.kt [hunk 1]
```diff
@@ -71,51 +75,51 @@ class RecallAuditService(
...
+  private fun getFieldPathForColumn(columnName: ColumnName): FieldPath =
     Recall::class.java.declaredFields.flatMap { field ->
-      val propCols = persister.getPropertyColumnNames(field.name).map { ColumnName(it) to FieldName(field.name) }
+      val propCols = persister.getPropertyColumnNames(field.name).map { ColumnName(it) to FieldPath(field.name) }
       if (propCols.size == 1) {
         propCols
       } else {
         field.type.declaredFields.flatMap { childField ->
           val childFieldPath = "${field.name}.${childField.name}"
           val childProbCols = persister.getPropertyColumnNames(childFieldPath)
```
Comment (AndyNewton) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/service/RecallAuditService.kt:106]: `childPropCols` not `childProbCols`?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/component/RecallAuditComponentTest.kt [hunk 2]
```diff
@@ -147,7 +148,9 @@ class RecallAuditComponentTest : ComponentTestBase() {
     // assert that audit still works when sentencing info - which has nested values - is updated
     authenticatedClient.updateRecall(recallId, UpdateRecallRequest(sentenceLength = Api.SentenceLength(3, 0, 0), sentenceDate = today, licenceExpiryDate = today, indexOffence = "Offence 1", sentencingCourt = CourtId("ACCRYC"), sentenceExpiryDate = today))
```
Comment (AndyNewton) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/component/RecallAuditComponentTest.kt:149]: This and similar length lines could use wrapping/re-formatting.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/component/RecallAuditComponentTest.kt [hunk 3]
```diff
@@ -33,19 +34,19 @@ class RecallAuditComponentTest : ComponentTestBase() {
...
-      Arguments.of(FieldName("sentencingInfo.sentenceExpiryDate"), "nested LocalDate", UpdateRecallRequest(sentenceLength = Api.SentenceLength(3, 0, 0), sentenceDate = today, licenceExpiryDate = today, indexOffence = "Offence 1", sentencingCourt = CourtId("ACCRYC"), sentenceExpiryDate = today), today.toString()),
+      Arguments.of(FieldPath("currentPrison"), "String or Enum", UpdateRecallRequest(currentPrison = PrisonId("ABC")), "ABC"),
+      Arguments.of(FieldPath("contraband"), "Boolean", UpdateRecallRequest(contraband = true), true),
+      Arguments.of(FieldPath("lastReleaseDate"), "LocalDate", UpdateRecallRequest(lastReleaseDate = today), today.toString()),
+      Arguments.of(FieldPath("assessedByUserId"), "UUID", UpdateRecallRequest(assessedByUserId = authenticatedClient.userId), authenticatedClient.userId.toString()),
+      Arguments.of(FieldPath("sentencingInfo.sentenceLength.sentenceYears"), "nested Integer", UpdateRecallRequest(sentenceLength = Api.SentenceLength(3, 0, 0), sentenceDate = today, licenceExpiryDate = today, indexOffence = "Offence 1", sentencingCourt = CourtId("ACCRYC"), sentenceExpiryDate = today), 3),
```
Comment (AndyNewton) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/component/RecallAuditComponentTest.kt:41]: Too wide to read effectively! 🙏

**Review by AndyNewton — APPROVED** [review]
Approved.
Couple of minor comments - but mainly I am concerned about how we'll support this through model change ... a subject for discussion outside of a PR though.

File: src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/service/RecallAuditService.kt (hunk 1 above)
Comment (philsegal) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/service/RecallAuditService.kt:106]: child PropertyColumns (from getPropertyColumnNames)

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/component/RecallAuditComponentTest.kt (hunk 2 above)
Comment (philsegal) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/component/RecallAuditComponentTest.kt:149]: I actually like that they are a single line 😬

File: src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/component/RecallAuditComponentTest.kt (hunk 3 above)
Comment (philsegal) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/managerecallsapi/component/RecallAuditComponentTest.kt:41]: Ditto :) . I prefer to see that each iteration is 1 line and 'easily' compare 2 lines. but can see some prefer a wrapped version.

**Pull request merged**